### PR TITLE
Use `Literal` types in `argparse`

### PR DESCRIPTION
**Pull request by sobolevn**

### CODE REVIEW

File: stdlib/argparse.pyi [hunk 1]
```diff
@@ -1,17 +1,19 @@
 import sys
 from typing import IO, Any, Callable, Generator, Generic, Iterable, NoReturn, Pattern, Protocol, Sequence, Type, TypeVar, overload
 
+from typing_extension import Literal
```
Comment (AlexWaygood) [inline stdlib/argparse.pyi:4]:
```suggestion
from typing_extensions import Literal
```

**Comment by sobolevn** [comment]
🤦

**Comment by Akuli** [comment]
`argparse` is trickier than you might expect. It uses `is` to compare strings, so there is a difference between passing `argparse.SUPPRESS` and passing `"==SUPPRESS=="`:

```
akuli@akuli-desktop:~$ python3 -c 'import argparse; p = argparse.ArgumentParser(); p.add_argument("--xyz", help=argparse.SUPPRESS); p.parse_args()' --help
usage: -c [-h]

optional arguments:
  -h, --help  show this help message and exit

akuli@akuli-desktop:~$ python3 -c 'import argparse; p = argparse.ArgumentParser(); p.add_argument("--xyz", help="==SUPPRESS=="); p.parse_args()' --help
usage: -c [-h] [--xyz XYZ]

optional arguments:
  -h, --help  show this help message and exit
  --xyz XYZ   ==SUPPRESS==
```

With this in mind, I'm not sure if `Literal` is the correct type for these constants. They really should be treated as if they were `typing.NewType` or something.

**Comment by sobolevn** [comment]
@Akuli thanks for the interesting use-case, I was not aware of that 🙂 

But, I still think that `Literal` is the correct type here. 
Because, I am not `add_argument` or `help`, I am just fixing constants to be in-line with https://typing.readthedocs.io/en/latest/source/stubs.html#constants

**Comment by Akuli** [comment]
But the constants really aren't meant to be treated as strings, but as special sentinel objects. `Literal["foo"]` means that you might as well pass the string `"foo"`, and there should be no difference.

**Comment by JelleZijlstra** [comment]
Closing per @Akuli's comment; there is also a merge conflict.

**Pull request closed without merging**